### PR TITLE
Kotlin Convenience Class Try 2

### PR DESCRIPTION
**Pull request by creativedrewy**

### CODE REVIEW

File: android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                    Log.e(TAG, "IO error while sending operation", cause)
+                    throw e
+                }
+                is TimeoutException -> {
+                    Log.e(TAG, "Timed out while waiting for result", cause)
+                    throw e
+                }
+                is JsonRpc20Client.JsonRpc20RemoteException -> {
+                    when (cause.code) {
+                        ProtocolContract.ERROR_REAUTHORIZE -> Log.e(TAG, "Reauthorization required", cause)
```
Comment (olivierg13) [inline android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt:68]: That's the one we might want to automate (i.e. call reauthorize())?

File: android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.concurrent.TimeoutException
+
+class MobileWalletAdapter(
+    private val timeout: Int = Scenario.DEFAULT_CLIENT_TIMEOUT_MS,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+) {
+
+    private val adapterOperations = LocalAdapterOperations(ioDispatcher)
+
+    suspend fun <T> transact(sender: ActivityResultSender, block: suspend AdapterOperations.() -> T) {
```
Comment (steveluscher) [inline android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt]: What would this API look like for different association scenarios other than `LocalAssociationScenario`?

```
// Basic, local.
transact(sender) { it.authorize(...) }

// Remote?
transact.withRemoteWallet(sender) { it.authorize(...) }
```

Or would it be more like…

```
transact(scenario, sender) { it.authorize(...) }
```

File: android/clientlib-ktx/build.gradle [hunk 3]
```diff
@@ -29,11 +29,16 @@ android {
...
+
+    buildFeatures {
+        buildConfig = false
+    }
 }
 
 dependencies {
     api(project(":clientlib"))
     implementation 'androidx.core:core-ktx:1.8.0'
+    implementation "androidx.activity:activity-ktx:1.5.0"
```
Comment (olivierg13) [inline android/clientlib-ktx/build.gradle:41]: You might be able to delete this dependency now with the `ActivityResultSender`?

File: android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/AdapterOperations.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.solana.mobilewalletadapter.clientlib
+
+import android.net.Uri
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient
+
+interface AdapterOperations {
```
Comment (olivierg13) [inline android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/AdapterOperations.kt:6]: This could live in the core clientlib?

**Review by olivierg13 — APPROVED** [review]
Looks great! 🎉

**Pull request merged**